src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -155,20 +155,25 @@ private void RequestRefresh(AuthenticationTicket ticket)
                 return result;
             }
 
-            var context = new CookieValidatePrincipalContext(Context, Scheme, result.Ticket, Options);
+            var context = new CookieValidatePrincipalContext(Context, Scheme, Options, result.Ticket);
             await Events.ValidatePrincipal(context);
 
-            if (context.Principal == null)
+            if (context.ShouldRenew && context.Ticket != null)
```

Comment:
I'll file a follow up bug to look into normalizing cookies (adding concept of skip across the board)